File: csharp/7-kyu/what-dominates-your-array/solution.cs
using System;
using System.Linq;
public class Kata
{
  public static int WhoDominates(int[] array)
  {
    Array.Sort(array);
    var value = 0;
    var count = 0;
    for (var i = 0; i < array.Length; i++)
    {
      if (value == array[i])
      {
        count++;
      }
      else
      {
        value = array[i];
        count = 1;
      }
      if (count > array.Length / 2)
      {
        return value;
      }
    }
    return -1;
  }
}

File: javascript/7-kyu/what-dominates-your-array/solution.js
function dominator(arr) {
  arr.sort();
  let value = 0;
  let count = 0;
  for (var i = 0; i < arr.length; i++) {
    if (value == arr[i]) {
      count++;
    } else {
      value = arr[i];
      count = 1;
    }
    if (count > arr.length / 2) return value;
  }
  return -1;
}
